app/models/concerns/hostext/ownership.rb

```diff
@@ -0,0 +1,67 @@
+module Hostext
+  module Ownership
+    extend ActiveSupport::Concern
+
+    included do
+      OWNER_TYPES = %w(User Usergroup)
+      belongs_to :owner, :polymorphic => true
+
+      before_validation :set_default_user
+      validates :owner_type, :inclusion =>
+                               {:in => OWNER_TYPES,
+                               :allow_blank => true,
+                               :message     => (_("Owner type needs to be one of the following: %s") % OWNER_TYPES.join(', '))}
+      validate :validate_owner
+      validate :owner_taxonomies_match, :if => Proc.new { |host| host.owner.is_a?(User) }
+    end
+
+    # method to return the correct owner list for host edit owner select dropbox
+    def is_owned_by
+      owner.id_and_type if owner
+    end
+
+    # virtual attributes which sets the owner based on the user selection
+    # supports a simple user, or a usergroup
+    # selection parameter is expected to be an ActiveRecord id_and_type method (see Foreman's AR extentions).
+    def is_owned_by=(selection)
+      oid = User.find(selection.to_i) if selection =~ (/-Users\Z/)
```

Comment:
This could also go into a `UserFactory`.

```ruby
self.owner = UserCollectionFactory.from_id_and_type(selection)
```

This makes this easier to test and would reduce code redundancy.